src/porepy/models/slightly_compressible_flow_model.py

```diff
@@ -109,63 +95,27 @@ def _assign_discretizations(self) -> None:
         """
 
         super()._assign_discretizations()
-        gb = self.gb
-        grid_list = [g for g, _ in gb.nodes()]
-
-
-        # Ad representation of discretizations
-        mass_mat = pp.MassMatrix(self.parameter_key)
-        mass_mat.discretize(gb)
-
-        # Ad variables
-        p = self._eq_manager.merge_variables([(g, self.variable) for g in grid_list])
-
-        # Update equations
-        subdomain_flow_eq = self._eq_manager.equations["subdomain_flow"]
-        interface_flow_eq = self._eq_manager.equations["interface_flow"]
-
-        mass = (
-            mass_mat * (p - p.previous_timestep()) / self.time_step
+        
+        if not hasattr(self, "dof_manager"):
+            self.dof_manager = pp.DofManager(self.gb)
+        
+        # Collection of subdomains
+        subdomains: List[pp.Grid] = [g for g, _ in self.gb]
+        
+        # AD representation of the mass operator
+        accumulation_term = pp.ad.MassMatrixAd(self.parameter_key, subdomains)
+        
+        # Need to find a way / refactor to access the superclass "and pressure" instead of create a new one
+        # If for some reason the mesh ordering change, this will be inconsistent
+        p = self._eq_manager.merge_variables([(g, self.variable) for g in subdomains])
+        time_step_ad = pp.ad.Scalar(self.time_step, "time step")
```

Comment:
Basic operations with floats and ints should work in combination with pp.ad.Operator (see Operator._parse_other)
No need for type conversion

Reply:
There actually is a reason for this: We want to be able to change the time step through updating time_step_ad._value